sdk/core/Azure.Core/src/Request.cs

```diff
@@ -2,48 +2,128 @@
 // Licensed under the MIT License.
 
 using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
-using System.Net.Http;
 using Azure.Core.Pipeline;
 
 namespace Azure.Core
 {
     /// <summary>
-    /// Represents an HTTP request. Use <see cref="HttpPipeline.CreateMessage()"/> or <see cref="HttpPipeline.CreateRequest"/> to create an instance.
+    /// Represents an HTTP request. Use <see cref="HttpPipeline.CreateMessage()"/>
+    /// or <see cref="HttpPipeline.CreateRequest"/> to create an instance.
     /// </summary>
 #pragma warning disable AZC0012 // Avoid single word type names
-    public abstract class Request : IDisposable
+    public abstract class Request : PipelineRequest
 #pragma warning restore AZC0012 // Avoid single word type names
     {
-        private RequestUriBuilder? _uri;
+        private RequestUriBuilder? _uriBuilder;
+        private string _method = RequestMethod.Get.Method;
+        private RequestContent? _content;
 
         /// <summary>
         /// Gets or sets and instance of <see cref="RequestUriBuilder"/> used to create the Uri.
         /// </summary>
-        public virtual RequestUriBuilder Uri
+        public new virtual RequestUriBuilder Uri
         {
-            get
-            {
-                return _uri ??= new RequestUriBuilder();
-            }
+            get => _uriBuilder ??= new RequestUriBuilder();
             set
             {
                 Argument.AssertNotNull(value, nameof(value));
-                _uri = value;
+                _uriBuilder = value;
             }
         }
 
         /// <summary>
         /// Gets or sets the request HTTP method.
         /// </summary>
-        public virtual RequestMethod Method { get; set; }
+        public new virtual RequestMethod Method
+        {
+            get => RequestMethod.Parse(MethodCore);
+            set => MethodCore = value.Method;
+        }
 
         /// <summary>
         /// Gets or sets the request content.
         /// </summary>
-        public virtual RequestContent? Content { get; set; }
+        public new virtual RequestContent? Content
+        {
+            get => (RequestContent?)ContentCore;
+            set => ContentCore = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the client request id that was sent to the server as <c>x-ms-client-request-id</c> headers.
+        /// </summary>
+        public abstract string ClientRequestId { get; set; }
+
+        /// <summary>
+        /// Gets the request HTTP headers.
+        /// </summary>
+        public new RequestHeaders Headers => new(this);
+
+        #region Overrides for "Core" methods from the PipelineRequest Template pattern
```

Comment:
![image](https://github.com/Azure/azure-sdk-for-net/assets/913445/da15c6d9-1d5e-48c8-8d3d-37d25cac4037)

Reply:
😢